Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsZipPreprocessExecuteAndRunUnitTestsWithDOMAndMochaExecutionStrategy.cs

```diff
@@ -0,0 +1,322 @@
+                                                    this.JsCodePreevaulationCode +
+                                                    this.JsCodeEvaluation +
+                                                    this.JsCodePostevaulationCode;
+
+        protected override string JsCodePreevaulationCode => @"
+chai.use(sinonChai);
+
+describe('TestDOMScope', function() {
+    before(function(done) {
+        jsdom.env({
+            html: '',
+            done: function(errors, window) {
+                global.window = window;
+                global.document = window.document;
+                global.$ = jq(window);
+                global.handlebars = handlebars;
+                Object.getOwnPropertyNames(window)
+                    .filter(function (prop) {
+                        return prop.toLowerCase().indexOf('html') >= 0;
+                    }).forEach(function (prop) {
+                        global[prop] = window[prop];
+                    });
+                done();
+            }
+        });
+    });
+
+	it('Test', function(done) {
+        this.timeout(30000);
+		var content = '';
+        var userBundleCode = '';
+            process.stdin.resume();
+            process.stdin.on('data', function (buf) {
+                content += buf.toString();
+            });
+            process.stdin.on('end', function () {
+                stream.on('data', function (x) {
+                    userBundleCode += x;
+                });
+                stream.on('end', afterBundling);
+                browserify(" + UserInputPlaceholder + @")
+                    .transform('" + this.BabelifyModulePath + @"', { plugins: ['" + this.EcmaScriptImportPluginPath + @"']})
+                    .bundle()
+                    .pipe(stream);";
+
+        protected override string JsCodeEvaluation => @"
+        function afterBundling()
+        {
+            var bgCoderConsole = {};
+            Object.keys(console)
+                .forEach(function (prop) {
+                    bgCoderConsole[prop] = console[prop];
+                    console[prop] = new Function('');
+                });"
+
+            + this.JsNodeDisableCode + @"
+
+            var testCode = content.trim();
+            var testFunc = new Function(" + this.TestFuncVariables + @",'code', testCode);
+            var result = testFunc.call({}," + this.TestFuncVariables.Replace("'", string.Empty) + @", userBundleCode);
+
+            Object.keys(bgCoderConsole)
+                .forEach(function (prop) {
+                    console[prop] = bgCoderConsole[prop];
+                });
+            done();
+        }";
+
+        protected override string JsCodePostevaulationCode => @"
+        });
+    });
+});";
+
+    public override ExecutionResult Execute(ExecutionContext executionContext)
+        {
+            ExecutionResult result = new ExecutionResult();
+
+            result.IsCompiledSuccessfully = true;
```

Comment:
you can skip this line if you use *object initializer* -> `var result = new ExecutionResult { IsCompiledSuccessfully = true }; `